src/Azure/Orleans.Transactions.AzureStorage/Storage/AzureTransactionLogStorage.cs

```diff
@@ -388,13 +398,15 @@ public ArchivalRow(ClusterOptions clusterOptions, byte[] transactions, long firs
             public byte[] Transactions { get; set; }
         }
 
-        private class CommitRow : TableEntity
+        public class CommitRow : TableEntity
         {
-            public CommitRow(long firstLSN, string serviceId)
+            public const string MaxRowKey = "crrk_~";
+            public const string MinRowKey = "crrk_";
+            public CommitRow(ClusterOptions clusterOptions, long firstLSN)
             {
                 // All entities are in the same partition for atomic read/writes.
-                PartitionKey = serviceId;
-                RowKey = ToRowKey(firstLSN);
+                PartitionKey = ArchivalRow.MakePartitionKey(clusterOptions.ServiceId);
```

Comment:
Yeah? I would think we would give these different partition keys

Reply:
Ok, so the reasoning is that we want to 'move' the rows in a batch transaction. That seems fine, though it makes me wonder what the purpose of this feature is.

If it's to improve startup speed (allowing more records to be skipped during recovery) then that's easily accomplished using just the StartRow. If it's so that we can retain the entire tx log for archival purposes then I would assume that we want to be able to move that log into some colder storage, or at least a different storage account.

So I am missing the motivation for this feature.

Reply:
We need a history of what transactions have succeeded.  Truncation normally removes completed transactions, losing history.

Reply:
So why not use the StartRow for this? Then you just have a pointer to the first "non-archived" transaction and you don't need to modify any of the records. It's faster, too.

Reply:
Given this is only indented for test cases, what's here is fine